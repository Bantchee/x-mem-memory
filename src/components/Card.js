const Card = (props) => {
    return (
        <button
            className="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400" 
        >
            <img 
                className="rounded-xl w-28 h-36"
                src={props.imgUrl} alt={props.name} 
            />
            <p 
            className="text-neutral-900 bottom-0 text-center font-bold text-base"
            >
                {props.name}
            </p>
        </button>
    )
};

export default Card;