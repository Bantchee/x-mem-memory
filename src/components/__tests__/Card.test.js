import Card from "../Card";
import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";

describe("Card Component", () => {
    test("render card component", () => {
        const { container } = render(<Card />);
        expect(container).toMatchSnapshot();
    });

    test("calls onClick when card is clicked", () => {
        const onClickMock = jest.fn();
        render(<Card onClick={onClickMock} />);
        const card = screen.getByRole("button");

        userEvent.click(card);

        expect(onClickMock).toHaveBeenCalled();
    });
});