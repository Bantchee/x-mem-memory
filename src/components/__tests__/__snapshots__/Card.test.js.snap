// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component render card component 1`] = `
<div>
  <button
    class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
  >
    <img
      class="rounded-xl w-28 h-36"
    />
    <p
      class="text-neutral-900 bottom-0 text-center font-bold text-base"
    />
  </button>
</div>
`;
